src/main/java/org/swisspush/redisques/RedisQues.java

```diff
@@ -124,13 +125,14 @@ private String getQueueCheckLastexecKey() {
         final String queue = event.body();
         log.debug("RedisQues Got registration request for queue " + queue + " from consumer: " + uid);
         // Try to register for this queue
-        redisClient.setnx(getConsumersPrefix() + queue, uid, event1 -> {
+        SetOptions setOptions = new SetOptions().setNX(true).setEX(2 * refreshPeriod);
```

Comment:
could you combine the expiry value with the _refreshRegistration_ method. Now we have two places in the code where we calculate the expiry value by calling

```java
2 * refreshPeriod
```
maybe also improve the _refreshRegistration_ method, since there the mentioned calculation is made 3 times :-)